File: src/main/java/com/fish/exception/ServiceException.java
package com.fish.exception;

import lombok.Getter;

@Getter
public class ServiceException extends RuntimeException{
    private final String msg;
    private final int code;
    public ServiceException(ServiceExceptionEnum serviceExceptionEnum) {
        this.msg= serviceExceptionEnum.getMsg();
        this.code= serviceExceptionEnum.getCode();
    }
    public ServiceException(int code, String message) {
        this.msg= message;
        this.code= code;
    }
}

File: src/main/kotlin/com/fish/exception/ServiceException.kt
package com.fish.exception

class ServiceException : RuntimeException {
    val msg: String
    val code: Int
    constructor(serviceExceptionEnum: ServiceExceptionEnum) {
        msg = serviceExceptionEnum.msg
        this.code = serviceExceptionEnum.code
    }
    constructor(code: Int, message: String) {
        msg = message
        this.code = code
    }
}
